### PR TITLE
[actpool] forbid processing system action

### PR DESCRIPTION
**Pull request by dustinxie**
# Description
Please include a summary of the changes and the related issue. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes #(issue)

## Type of change
Please delete options that are not relevant.
- [] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [] Code refactor or improvement
- [] Breaking change (fix or feature that would cause a new or changed behavior of existing functionality)
- [] This change requires a documentation update

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration
- [x] make test
- [] fullsync
- [] Other test (please specify)

**Test Configuration**:
- Firmware version:
- Hardware:
- Toolchain:
- SDK:

# Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: api/coreservice.go [hunk 1]
```diff
@@ -411,6 +411,11 @@ func (core *coreService) SendAction(ctx context.Context, in *iotextypes.Action)
 		return "", status.Error(codes.InvalidArgument, err.Error())
 	}
 
+	// system action is only added by proposer when creating a block
+	if action.IsSystemAction(selp) {
+		return "", errors.New("cannot process system action")
+	}
```
Comment (dustinxie) [inline api/coreservice.go:417]: to prevent potential replay of grantreward

**Review by Liuhaai — APPROVED** [review]

**Comment by CoderZhi** [comment]
https://github.com/iotexproject/iotex-core/blob/master/state/factory/workingset.go#L455

**Review by envestcc — APPROVED** [review]

**Review by CoderZhi — CHANGES_REQUESTED** [review]
It has been prevented in workingset

**Comment by dustinxie** [comment]
> It has been prevented in workingset

yes, but that's after a block is proposed (and then found invalid), so a block is wasted. In other words, someone can potentially keep sending such a system action and prevent block being generated, like a DoS attack?

so it's worthwhile to early-detect this and throw it out of action pool?

**Comment by CoderZhi** [comment]
> > It has been prevented in workingset
> 
> yes, but that's after a block is proposed (and then found invalid), so a block is wasted. In other words, someone can potentially keep sending such a system action and prevent block being generated, like a DoS attack?
> 
> so it's worthwhile to early-detect this and throw it out of action pool?

if so, we need to prevent adding it to actpool.

**Comment by dustinxie** [comment]
> > > It has been prevented in workingset
> > 
> > 
> > yes, but that's after a block is proposed (and then found invalid), so a block is wasted. In other words, someone can potentially keep sending such a system action and prevent block being generated, like a DoS attack?
> > so it's worthwhile to early-detect this and throw it out of action pool?
> 
> if so, we need to prevent adding it to actpool.

yes, by early-exiting here, system action is neither added to local actpool (L436 below), nor being propagated to p2p network (L460 below)

**Pull request merged**